core/src/main/java/jenkins/model/CoreEnvironmentContributor.java

```diff
@@ -41,6 +41,8 @@ public void buildEnvironmentFor(Run r, EnvVars env, TaskListener listener) throw
 
     @Override
     public void buildEnvironmentFor(Job j, EnvVars env, TaskListener listener) throws IOException, InterruptedException {
+        env.put("CI", "true");
```

Comment:
Please note: I purposely didn't update the [buildEnv.groovy](https://github.com/jenkinsci/jenkins/blob/master/core/src/main/resources/jenkins/model/CoreEnvironmentContributor/buildEnv.groovy) or [buildEnv.properties](https://github.com/jenkinsci/jenkins/blob/master/core/src/main/resources/jenkins/model/CoreEnvironmentContributor/buildEnv.properties) files since it's a "constant variable" and technically has no real semantic value. Itfyou think it's worth adding it, I can update this PR to do so.

Reply:
Would be better to add it, because it might override user-defined variables